### PR TITLE
Structural and line edit on concepts/hypertables

### PR DESCRIPTION
**Pull request by charislam**
# Description

Make conceptual information on hypertables and chunks more scannable and beginner-friendly with structural and line edits

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Partial #801

### CODE REVIEW

**Comment by charislam** [comment]
Draft of index and chunks diagram (need SME input)
indexes-chunks drawio

**Comment by charislam** [comment]
@ryanbooz I am rewriting the concept guides to be more scannable and beginner-friendly. Could you take a look? I also have a rough sketch of a diagram to explain local indexes, which I'm sure could do with a lot of improvements. Thanks!

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 1]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+partitioned by some other column in addition to time. This is sometimes called
+"time-and-space partitioning."
 
-Hypertables must be partitioned by time, but they may also be partitioned by
-another column. This is sometimes called "time-and-space" partitioning. 
+For example, say that you are collecting sensor data from multiple devices and
+storing it in a hypertable. You can partition the hypertable by both time and
+device ID. Instead of creating one chunk for each time interval, TimescaleDB
+creates multiple chunks for each time interval. It decides which chunk each row
+belongs in by looking at its time value *and* its device ID.
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:68]:
```suggestion
belongs in by looking at its time value and its device ID. 
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 2]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+are hypertables in multi-node databases, where the hypertable data is
+distributed among nodes. For each time interval, each node contains a portion of
+the data. This allows parallel inserts and queries. For example, TimescaleDB can
+write data about device A at time Z to chunk C on node M, at the same time as it
+writes data about device B at time Z to chunk D on node N.
+
+For more information, see the documentation on [distributed
+hypertables][distributed-hypertables].
 
 ## Chunk architecture
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:83]:
```suggestion
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 3]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
-based on the chunks' dimensions.  And queries to a hypertable use knowledge
-of these chunks' constraints to only "push down" a query to the proper
-chunks: If a query specifies that `time > now() - interval '1 week'`, for
-example, the database only executes the query against chunks covering
-the past week, and excludes any chunks before that time.  This happens
-transparently to the user, however, who simply queries the hypertable with
-a standard SQL statement.
+Each chunk is a standard PostgreSQL table. The hypertable is an abstraction on
+top of the chunks, making the whole entity look and behave like a single table.
+In PostgreSQL vocabulary, the hypertable is a "parent table" and the chunks are
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:86]: terminology?

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 4]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+its "child tables."
+
+To enforce each chunk's time boundaries, each chunk includes constraints. For
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:89]: includes *time* constraints?

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 5]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+The TimescaleDB planner uses knowledge of chunks' constraints to optimize
+database operations. When a row is inserted, it routes the insertion into the
+right chunk for the row's time value. When a query is made, it pushes the query
+down to only the chunks that matter. 
+
+For example, if a query has a `WHERE` clause specifying `time > now() - INTERVAL
+'1 week'`, the database only executes the query against chunks covering the past
+week. It excludes any chunks from before that time. 
+
+All of this happens in the background, transparently to the user.
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:102]: Third person.

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 6]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+`['2020-07-01 00:00:00+00', '2020-07-02 00:00:00+00']`.
+
+The TimescaleDB planner uses knowledge of chunks' constraints to optimize
+database operations. When a row is inserted, it routes the insertion into the
+right chunk for the row's time value. When a query is made, it pushes the query
+down to only the chunks that matter. 
+
+For example, if a query has a `WHERE` clause specifying `time > now() - INTERVAL
+'1 week'`, the database only executes the query against chunks covering the past
+week. It excludes any chunks from before that time.
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:100]: This section feels like it needs fewer para breaks.

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 7]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+down to only the chunks that matter. 
+
+For example, if a query has a `WHERE` clause specifying `time > now() - INTERVAL
+'1 week'`, the database only executes the query against chunks covering the past
+week. It excludes any chunks from before that time. 
+
+All of this happens in the background, transparently to the user.
+
+### Space-partitioning architecture
+
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:105]:
```suggestion
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md [hunk 8]
```diff
@@ -40,64 +40,82 @@ In addition to time, you can also partition a hypertable by a space parameter.
...
+Space partitions are also enforced with chunk constraints. Partitions are always
+non-overlapping.
+
+To determine how the values of the space-parititioning column map to partitions,
+two methods can be used:
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks.md:110]:
```suggestion
choose from one of these two methods:
```

**Review by Loquacity — CHANGES_REQUESTED** [review]

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md (hunk 3 above)
Comment (charislam) [reply timescaledb/overview/core-concepts/hypertables-and-chunks.md:86]:
```suggestion
In PostgreSQL terminology, the hypertable is a "parent table" and the chunks are
```
```suggestion
In PostgreSQL vocabulary, the hypertable is a "parent table" and the chunks are
```

Comment (charislam) [reply timescaledb/overview/core-concepts/hypertables-and-chunks.md:86]:
```suggestion
In PostgreSQL terminology, the hypertable is a "parent table" and the chunks are
```
```suggestion
In PostgreSQL vocabulary, the hypertable is a "parent table" and the chunks are
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks.md (hunk 4 above)
Comment (charislam) [reply timescaledb/overview/core-concepts/hypertables-and-chunks.md:89]:
```suggestion
To enforce each chunk's time boundaries, each chunk includes time constraints. For
```
```suggestion
To enforce each chunk's time boundaries, each chunk includes constraints. For
```

**Review by davidkohn88 — COMMENTED** [review]
Looking good generally, I left a few comments. I think the section on indexes is a little hard to grok and could be a good place for a few diagrams.

File: timescaledb/overview/core-concepts/hypertables-and-chunks-benefits.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+Even with multiple local indexes, TimescaleDB can still ensure global uniqueness
+for keys. It enforces an important constraint: any key that requires uniqueness,
+such as a `PRIMARY KEY`, must include all columns that are used for data
+partitioning.
+
+In other words, because data is partitioned between chunks based on time value,
+the unique key must include the time value. When data is inserted, TimescaleDB
+identifies the corresponding time chunk and checks for uniqueness within that
+chunk. Because no other chunk can contain that time value, uniqueness within the
+chunk implies global uniqueness.
```
[the remark this answers is not in the document]
Comment (charislam) [reply timescaledb/overview/core-concepts/hypertables-and-chunks-benefits.md]: I took a stab at making diagrams for this, would love some feedback here! https://iobeam.slack.com/archives/C0360448V6X/p1647371342405649

File: timescaledb/overview/core-concepts/hypertables-and-chunks/chunk-architecture.md [hunk 10]
```diff
@@ -20,15 +20,53 @@ All of this happens in the background. From your perspective, the hypertable
...
 choose from one of these two methods:
 *   Hashing, which maps each value into one of several defined hash buckets.
     This is the most common method.
 *   Interval-based partitioning. This works the same way as time-based
     partitioning, where all the values in a specified range fall into the same
     partition.
 
-## Chunk local indexes
\ No newline at end of file
+Like time partitions, space partitions are always non-overlapping.
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/chunk-architecture.md:33]:
```suggestion
Like time partitions, space partitions never overlap.
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md [hunk 11]
```diff
@@ -10,11 +10,13 @@ ago. The exact timescales depend on your application.
...
 So, when you insert and query recent data, these operations access data from
-memory, which is faster than loading data from disk.
+memory. This improves operation speed, since data doesn't need to be loaded from
+disk.
 
-To get the best performance, the most recent chunk, including indexes, should be
-no larger than 25% of your system's main memory. For more information, see the
-documentation on [chunk sizing][chunk-sizing].
+<highlight type="note">
+To learn about chunk sizing for improved performance, see the section on
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md:17]:
```suggestion
For more information about chunk sizing for improved performance, see the section on 
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/index.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+In addition to time, you can also partition a hypertable by space parameters.
+For more information, see the section on [time-and-space
+partitioning](#hypertables-can-also-be-partitioned-by-space).
+
+<highlight type="note">
+This section uses the example of 1-day chunks to explain how hypertables work.
+The default chunk time interval is actually 7 days, and you can change it to
+optimize partitioning for your data. To learn about best practices for chunk 
+time intervals, see the documentation on [chunk sizing](FIXME).
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/index.md:44]:
```suggestion
time intervals, see the documentation on [chunk sizing](timescaledb/latest/how-to-guides/hypertables/best-practices/#time-intervals).
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/chunk-architecture.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    partition.
+
+Like time partitions, space partitions never overlap.
+
+## Chunk local indexes
+Rather than building a global index over an entire hypertable, TimescaleDB
+builds local indexes on each chunk. In other words, each chunk has its own index
+that only indexes data within that chunk. This optimization improves insert
+speed for recent data. For more information, see the section on the [benefits of
+local indexes][hypertable-benefits-indexes].
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/chunk-architecture.md:40]:
```suggestion
speed for recent data. For more information, see the section on the 
[benefits of local indexes][hypertable-benefits-indexes].
```

**Review by Loquacity — CHANGES_REQUESTED** [review]
Just noticed a bunch of links broken across lines.

**Review by Loquacity — APPROVED** [review]

File: timescaledb/overview/core-concepts/hypertables-and-chunks/index.md [hunk 14]
```diff
@@ -1,93 +1,40 @@
...
-hypertables][distributed-hypertables].
-
-[chunk-sizing]: /how-to-guides/hypertables/best-practices/#time-intervals
-[create-hypertable]: /how-to-guides/hypertables/create/
-[distributed-hypertables]: /overview/core-concepts/distributed-hypertables/
-[hypertable-benefits]: /overview/core-concepts/hypertables-and-chunks-benefits/
+Hypertables are PostgreSQL tables with special features that make it easy to
+handle time-series data. Anything you can do with a regular PostgreSQL table,
+you can do with a hypertable. In addition, you get the benefits of improved
+performance and user experience for time series.
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/index.md:5]:
```suggestion
performance and user experience for time series data.
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertable-architecture.md [hunk 15]
```diff
@@ -20,94 +24,74 @@ alt="Rows are inserted into hypertable chunks based on their time values." />
...
-[benefits of hypertables][hypertable-benefits].
-
-## Hypertables can also be partitioned by space
-All TimescaleDB hypertables are partitioned by time. They can also be
-partitioned by other columns in addition to time. This is sometimes called
-"time-and-space partitioning."
+## Time-and-space partitioning
+All TimescaleDB hypertables are partitioned by time. In addition, they might
+also be partitioned by other columns. This is called "time-and-space
+partitioning."
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertable-architecture.md:38]:
```suggestion
also be partitioned by other columns. This is called time-and-space
partitioning.
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md [hunk 16]
```diff
@@ -83,14 +84,15 @@ data must be migrated to the new server when you expand a cluster.
...
+immediate data migration.
+
+The TimescaleDB planner handles queries seamlessly as partitions change. Behind
+the scenes, it keeps track of where each chunk is stored.
 
 To fine-tune server rebalancing, you can asynchronously migrate chunks. Or, you
 can delete older data by using data retention policies. Older data is
 partitioned according to the older server setup, while newer data is partitioned
-according to the newer setup. So data retention eventually rebalances data
+according to the newer setup. Thus, data retention eventually rebalances data
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md:95]: "Thus"?

File: timescaledb/overview/core-concepts/hypertables-and-chunks/index.md [hunk 17]
```diff
@@ -34,6 +34,14 @@ For example, you might have:
 *   A hypertable to record truck GPS coordinates over time, and a regular table
     to record the make, model, and age of each truck
 
+## Why can't you partition your time-series data yourself?
+You can! You can plan and write all the partitioning logic yourself on regular
```
Comment (Loquacity) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/index.md:38]: Not sure I like this construction 😕

File: timescaledb/overview/core-concepts/hypertables-and-chunks/index.md [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+Learn more about:
+*   The [benefits of hypertables][hypertable-benefits] for handling time-series
+    data
+*   [Hypertable architecture][hypertable-architecture], and how hypertables and
+    chunks work behind the scenes
+*   [How to work with hypertables][hypertable-how-to]
+
+## Hypertables compared to partitioning in regular PostgreSQL
+If you use regular PostgreSQL, you can also partition your time-series data by
+writing the partitioning logic yourself. But handling child tables, constraints,
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/index.md:26]: @Loquacity is this better?

**Review by Loquacity — APPROVED** [review]

**Comment by Loquacity** [comment]
@charislam I think that this can be merged, when you're ready.

**Comment by davidkohn88** [comment]
Did you want another look from me on this? I think we're ready to merge? I just keep getting pinged about it...

**Comment by charislam** [comment]
Sorry @davidkohn88, I think we can just go ahead and merge, I'll do that now to stop the constant notifs :)

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertable-architecture.md [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+command, and the correct index is automatically updated.
+
+<highlight type="note">
+To see examples of the partitioning column and unique index constraint in
+practice, see the section on
+[hypertables and unique indexes](/timescaledb/latest/how-to-guides/hypertables/hypertables-and-unique-indexes/).
+</highlight>
+
+[distributed-hypertables]: /overview/core-concepts/distributed-hypertables/
+[hypertable-benefits-indexes]: /overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits/#faster-index-updates
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertable-architecture.md:131]:
```suggestion
[distributed-hypertables]: /timescaledb/:currentVersion:/overview/core-concepts/distributed-hypertables/
[hypertable-benefits-indexes]: /timescaledb/:currentVersion:/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits/#faster-index-updates
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+This allows you to:
+*   Increase availability with multiple replicas
+*   Recover from node failure or primary server outage
+*   Horizontally scale reads by spreading query volume across multiple nodes
+
+For more information, see the [replication][replication] section.
+
+[chunk-migration]: /api/:currentVersion:/distributed-hypertables/move_chunk_experimental/
+[data-retention]: /how-to-guides/data-retention/
+[local-indexes]: /overview/core-concepts/hypertables-and-chunks/chunk-architecture/#chunk-local-indexes
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md:112]:
```suggestion
[data-retention]: /timescaledb/:currentVersion:/how-to-guides/data-retention/
[local-indexes]: /timescaledb/:currentVersion:/overview/core-concepts/hypertables-and-chunks/chunk-architecture/#chunk-local-indexes
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+[performance-benchmark]: https://www.timescale.com/blog/timescaledb-vs-6a696248104e/
+[replication]: /how-to-guides/replication-and-ha/
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits.md:114]:
```suggestion
[replication]: /timescaledb/:currentVersion:/how-to-guides/replication-and-ha/
```

File: timescaledb/overview/core-concepts/hypertables-and-chunks/index.md [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+*   A hypertable to record sensor readings over time, and a regular table
+    to record sensor location and other metadata
+*   A hypertable to record stock asset prices over time, and a regular table to
+    record ticker symbols and names for each asset
+*   A hypertable to record truck GPS coordinates over time, and a regular table
+    to record the make, model, and age of each truck
+
+[hypertable-architecture]: /overview/core-concepts/hypertables-and-chunks/hypertable-architecture/
+[hypertable-benefits]: /overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits/
+[hypertable-how-to]: /how-to-guides/hypertables/
```
Comment (charislam) [inline timescaledb/overview/core-concepts/hypertables-and-chunks/index.md:48]:
```suggestion
[hypertable-architecture]: /timescaledb/:currentVersion:/overview/core-concepts/hypertables-and-chunks/hypertable-architecture/
[hypertable-benefits]: /timescaledb/:currentVersion:/overview/core-concepts/hypertables-and-chunks/hypertables-and-chunks-benefits/
[hypertable-how-to]: /how-to-guides/hypertables/
```

**Pull request merged**